core/Repository/Credentials/IRepositoryDerivedCredentials.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cmf.CLI.Core.Repository.Credentials
+{
+    public interface IRepositoryDerivedCredentials
```

Comment:
So I figured what you are doing with this, it's fine, I just don't like the name. It seems any credential set that implements this interface would be a derived credential (I'd expect to see this applied to e.g. a cmf.io NPM credential set) but it's actually applied in the source credential of the derived credentials. Not wanting to nitpick much, but the name really threw me off. Also, we should document all of these interfaces to help with that as well.

Reply:
Yes, you are totally right, the naming was not the best here (to say the least). I renamed the interface and added a small description to explain what it means and the use case behind it.